std/file.d

```diff
@@ -821,27 +820,9 @@ version(Posix) private void writeImpl(const(char)[] name, const(FSChar)* namez,
                        : O_CREAT | O_WRONLY | O_TRUNC;
 
     immutable fd = core.sys.posix.fcntl.open(namez, mode, octal!666);
-    cenforce(fd != -1, name, namez);
-    {
-        scope(failure) core.sys.posix.unistd.close(fd);
-
-        immutable size = buffer.length;
-        size_t sum, cnt = void;
-        while (sum != size)
-        {
-            cnt = (size - sum < 2^^30) ? (size - sum) : 2^^30;
-            const numwritten = core.sys.posix.unistd.write(fd, buffer.ptr + sum, cnt);
-            if (numwritten != cnt)
-                break;
-            sum += numwritten;
-        }
-        cenforce(sum == size, name, namez);
-    }
-    cenforce(core.sys.posix.unistd.close(fd) == 0, name, namez);
+    writeToOpenFile(fd, buffer, name, namez);
```

Comment:
who closes fd?

Reply:
Looking at the code, `writeToOpenFile` does. I suppose that I could rename it to `writeToOpenFileAndClose` or something to make it clearer. It's just a private function that was created to refactor some code that `writeImpl` uses rather than duplicating it.